package proto_ff;

import "yd_fieldoptions.proto";


message E_FishRoom
{
	optional int32 m_id = 1[(yd_fieldoptions.field_cname) = "唯一ID"];
	optional int32 m_roomid = 2[(yd_fieldoptions.field_cname) = "房间ID"];
	optional int32 m_gameid = 3[(yd_fieldoptions.field_cname) = "游戏ID"];
	optional string m_gamename = 4[(yd_fieldoptions.field_cname) = "游戏名", (yd_fieldoptions.field_bufsize) = 64];
	optional int32 m_roomtype = 5[(yd_fieldoptions.field_cname) = "房间类型"];
	optional int32 m_roomlevel = 6[(yd_fieldoptions.field_cname) = "房间等级"];
	optional string m_roomname = 7[(yd_fieldoptions.field_cname) = "房间名", (yd_fieldoptions.field_bufsize) = 64];
	optional int32 m_sitenum = 8[(yd_fieldoptions.field_cname) = "座位数"];
	optional int32 m_deskcount = 9[(yd_fieldoptions.field_cname) = "桌子数"];
	optional int32 m_maxpeople = 10[(yd_fieldoptions.field_cname) = "最大玩家数"];
	optional int32 m_entermin = 11[(yd_fieldoptions.field_cname) = "最小筹码"];
	optional int32 m_entermax = 12[(yd_fieldoptions.field_cname) = "最大筹码"];
	optional int32 m_autochair = 13[(yd_fieldoptions.field_cname) = "选位置"];
	optional int32 m_isexpscene = 14[(yd_fieldoptions.field_cname) = "体验场"];
	optional int32 m_expscenegold = 15[(yd_fieldoptions.field_cname) = "体验场金币"];
	optional int32 m_tax = 16[(yd_fieldoptions.field_cname) = "抽水率"];
}


message Sheet_FishRoom
{
	repeated E_FishRoom E_FishRoom_List = 1[(yd_fieldoptions.field_arysize)=8];
}

message E_FishGunvalueYDesc
{
	optional int32 m_num = 1[(yd_fieldoptions.field_cname) = "num"];
	optional int32 m_id = 2[(yd_fieldoptions.field_cname) = "id"];
}

message E_FishGunvalueItemDesc
{
	optional int32 m_num = 1[(yd_fieldoptions.field_cname) = "num"];
	optional int32 m_item = 2[(yd_fieldoptions.field_cname) = ""];
}

message E_FishGunvalue
{
	optional int32 m_id = 1[(yd_fieldoptions.field_cname) = "唯一ID"];
	optional int32 m_value = 2[(yd_fieldoptions.field_cname) = "值"];
	optional int32 m_gameid = 3[(yd_fieldoptions.field_cname) = "游戏Id"];
	optional int32 m_roomid = 4[(yd_fieldoptions.field_cname) = "房间ID"];
	optional int32 m_gunid = 5[(yd_fieldoptions.field_cname) = "炮Id"];
	optional int32 m_gameroomid = 6[(yd_fieldoptions.field_cname) = "游戏房间ID"];
	repeated int32 m_x_id = 7[(yd_fieldoptions.field_cname) = "游房", (yd_fieldoptions.field_arysize) = 3];
	repeated E_FishGunvalueYDesc m_y = 8[(yd_fieldoptions.field_cname) = "游房间", (yd_fieldoptions.field_arysize) = 2];
	repeated E_FishGunvalueItemDesc m_item = 9[(yd_fieldoptions.field_cname) = "", (yd_fieldoptions.field_arysize) = 2];
}


message Sheet_FishGunvalue
{
	repeated E_FishGunvalue E_FishGunvalue_List = 1[(yd_fieldoptions.field_arysize)=64];
}

message E_FishCtrllevel
{
	optional int32 m_controllevel = 1[(yd_fieldoptions.field_cname) = "唯一ID"];
	optional int32 m_controlrate = 2[(yd_fieldoptions.field_cname) = "控制概率"];
}


message Sheet_FishCtrllevel
{
	repeated E_FishCtrllevel E_FishCtrllevel_List = 1[(yd_fieldoptions.field_arysize)=16];
}

message E_FishConfig
{
	optional int32 m_fishid = 1[(yd_fieldoptions.field_cname) = "鱼Id"];
	optional int32 m_fishtype = 2[(yd_fieldoptions.field_cname) = "鱼类型"];
	optional int32 m_buildfishtype = 3[(yd_fieldoptions.field_cname) = "鱼build类型"];
	optional int32 m_ratiomin = 4[(yd_fieldoptions.field_cname) = "最新比例"];
	optional int32 m_ratiomax = 5[(yd_fieldoptions.field_cname) = "最大比例"];
	optional int32 m_doubleawardminratio = 6[(yd_fieldoptions.field_cname) = "双重奖励最小比率"];
	optional int32 m_childfishcount = 7[(yd_fieldoptions.field_cname) = "子鱼数"];
	optional int32 m_damageradius = 8[(yd_fieldoptions.field_cname) = "伤害半径"];
	optional string m_damagefishids = 9[(yd_fieldoptions.field_cname) = "伤害子鱼Ids", (yd_fieldoptions.field_bufsize) = 128];
	repeated string m_childfishids = 10[(yd_fieldoptions.field_cname) = "子鱼Ids", (yd_fieldoptions.field_arysize) = 6, (yd_fieldoptions.field_bufsize) = 64];
}


message Sheet_FishConfig
{
	repeated E_FishConfig E_FishConfig_List = 1[(yd_fieldoptions.field_arysize)=64];
}
